### PR TITLE
Add `RUNTIME_ENCLAVE_BINARY_PATH` environment variable (2/2)

### PR DESCRIPTION
**Pull request by gbryant-arm**
Fix for #372

### CODE REVIEW

**Comment by dominic-mulligan-arm** [comment]
@mathias-arm can you take a look at this and see whether it fixes your concerns?

File: workspaces/linux-host/Makefile
Comment (mathias-arm) [inline workspaces/linux-host/Makefile:32]: With the changes in #339, the file `$(WORKSPACE_DIR)/linux-runtime/runtime_manager_enclave` will not be created anymore. This line should be:
```
	RUNTIME_ENCLAVE_BINARY_PATH=$(RUNTIME_MANAGER_ENCLAVE)
```

**Review by mathias-arm — CHANGES_REQUESTED** [review]

File: workspaces/linux-host/Makefile
Comment (gbryant-arm) [reply workspaces/linux-host/Makefile:32]: Good catch. Looks like not having `runtime_manager_enclave` breaks the tests.
Let's keep copying it from `./target`.
Making changes to #339

Comment (gbryant-arm) [reply workspaces/linux-host/Makefile:32]: Fixed in #339

**Comment by mathias-arm** [comment]
It seems the proper fix without a superfluous copy can't be implemented and tested in two separate PRs (#375 and #339). The easiest option is probably to move this change into #339.

**Comment by gbryant-arm** [comment]
> It seems the proper fix without a superfluous copy can't be implemented and tested in two separate PRs (#375 and #339). The easiest option is probably to move this change into #339.

Yeah there seems to be a bunch of unnecessary copies:
+ Nitro: `PCR0` copied to `css-nitro.bin`, `runtime_manager_enclave` copied to `docker`
+ Linux: `runtime_manager_enclave` copied to `./` and `./css-linux.bin`

I suggest we address that in another PR

**Comment by dominic-mulligan-arm** [comment]
What is the status of this PR?

**Comment by mathias-arm** [comment]
My suggestion is to close this PR, and work on this (and the additional copies) as part of #375.

**Comment by mathias-arm** [comment]
Sorry, yes I meant #339. I agree they are orthogonal, but they overlap a lot in what parts of the Makefile they modify. I don't think it would be possible to have two/three PRs that would not conflict. Regarding the copied files, I am worried that it might lead to using stale copies.

**Comment by gbryant-arm** [comment]
> Sorry, yes I meant #339. I agree they are orthogonal, but they overlap a lot in what parts of the Makefile they modify. I don't think it would be possible to have two/three PRs that would not conflict. Regarding the copied files, I am worried that it might lead to using stale copies.

#339 changes the runtime Makefile, #375 was changing the host Makefile. No overlaps.

I opened https://github.com/veracruz-project/veracruz/issues/383 for the copies, I'd like to keep my PRs self-contained if possible

**Comment by dominic-mulligan-arm** [comment]
Reopened until consensus emerges.

**Comment by dominic-mulligan-arm** [comment]
OK, trying to drive a consensus about what should be done with this PR.  From talking with @gbryant-arm:

- Apparently there is an issue affecting @egrimley-arm that this PR fixes.
- Moreover, the changes here are disjoint from #339 and @gbryant-arm wants to keep the PRs self-contained, and has opened an additional PR (#383) trying to resolve the copying issue.

@mathias-arm are you happy if this PR remains open, in light of the above?

**Comment by gbryant-arm** [comment]
I believe @egrimley-arm was affected by the issue described in https://github.com/veracruz-project/veracruz/pull/339/ instead

**Comment by egrimley-arm** [comment]
> I believe @egrimley-arm was affected by the issue described in #339 instead

Yes, that's right, I think.

**Comment by dominic-mulligan-arm** [comment]
What is the status of this PR?

**Comment by gbryant-arm** [comment]
> What is the status of this PR?

Waiting for @mathias-arm 's approval

**Comment by gbryant-arm** [comment]
Merging into #339

**Pull request closed without merging**
